tapdb/assets_store.go

```diff
@@ -1468,12 +1523,65 @@ func (a *AssetStore) ListEligibleCoins(
 		constraints,
 	})
 
-	// We only want to select unspent commitments.
+	// We only want to select unspent and non-leased commitments.
 	assetFilter.Spent = sqlBool(false)
+	assetFilter.Leased = sqlBool(false)
 
 	return a.queryCommitments(ctx, assetFilter)
 }
 
+// LeaseCoin leases/locks/reserves a coin for the given lease owner until the
+// given expiry. This is used to prevent multiple concurrent coin selection
+// attempts from selecting the same coin.
+func (a *AssetStore) LeaseCoin(ctx context.Context, utxoOutpoint wire.OutPoint,
```

Comment:
Hmm, I wonder if we want leasing at a lower level of abstractions: the full prev ID schema. I guess this works tho, since you can only ever spend that UTXO once. 

In the future, more granular leasing might be needed once we start to optimistically combine outgoing sends in the freighter.

Reply:
Makes sense. So we would either batch sends together in a timed manner. Or use RBF to replace a previously published anchoring TX? Either way, I think we'll need some more design around that and then probably go with a new column on the asset table itself.